weblog/server.go

```diff
@@ -2,49 +2,60 @@ package weblog
 
 import (
 	"fmt"
-	"log"
+	"net"
 	"net/http"
 
-	"github.com/gorilla/mux"
-
 	"github.com/deis/logger/storage"
 )
 
 const (
-	bindHost = "0.0.0.0"
-	bindPort = 8088
+	bindAddr = "0.0.0.0:8088"
 )
 
-// Server implements a simple HTTP server that handles GET and DELETE requests for application
-// logs.  These actions are accomplished by delegating to a storage.Adapter.
+// Server implements an HTTP server.
 type Server struct {
-	listening bool
-	router    *mux.Router
-	errCh     chan error
+	// The Listener used for incoming HTTP connections
+	Listener net.Listener
+	// Config may be changed before calling Start()
+	Config *http.Server
+	// base URL of form http://ipaddr:port with no trailing slash
+	URL string
 }
 
-// NewServer returns a pointer to a new Server instance.
-func NewServer(storageAdapter storage.Adapter) (*Server, error) {
-	return &Server{
-		router: newRouter(newRequestHandler(storageAdapter)),
-	}, nil
+// New returns a new HTTP Server and starts it.
+// The caller should call Close when finished to shut it down.
+func NewServer(storageAdapter storage.Adapter) *Server {
+	s := &Server{
+		Listener: defaultListener(),
+		Config:   &http.Server{Handler: newRouter(newRequestHandler(storageAdapter))},
+	}
+	s.Start()
```

Comment:
fwiw, I think it's odd (bad, IMO) to implicitly "start" things upon creation, especially when you require that thing to be explicitly _stopped_ / closed. It's easy to forget to defer a stop or close... it's easier to remember to do when you have to explicitly invoke start, listen, etc.